docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+            days_in_month = 30;
+        } else {
+            days_in_month = 31;
+        }
+        uint32_t seconds_in_month = days_in_month * seconds_per_day;
+        if (remaining_seconds < seconds_in_month) {
+            month = i;
+            break;
+        }
+        remaining_seconds -= seconds_in_month;
+        num_days += days_in_month;
+    }
+
+    // Calculate the day
+    day = remaining_seconds / seconds_per_day + 1;
+    remaining_seconds = remaining_seconds % seconds_per_day;
+
+    // Calculate the hour
+    hour = remaining_seconds / seconds_per_hour;
+    remaining_seconds = remaining_seconds % seconds_per_hour;
+
+    // Calculate the minute
+    minute = remaining_seconds / seconds_per_minute;
+
+    // Calculate the second
+    second = remaining_seconds % seconds_per_minute;
+}
+```
+
+Build the smart contract again and deploy it to the local node as you did previously.
+
+Now you can send an `hi` action to the local node and observe that both `hi` and `gettime` actions are executed.
+
+```shell
+dune --send-action hello hi '[ama]' hello@active
+```
+
+```txt
+executed transaction: 78ece3e63c3c6f11a210a9229f32b822c825fc72b268b1325b7dc1798797f460  104 bytes  431 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ] 
+#         hello <= hello::hi                    {"nm":"ama"}
+>> Name : ama
+#         hello <= hello::gettime               ""
+>> Year: 2023, Month: 2, Day: 24, Hour: 0, Minute: 42, Second: 5
+```
+
+## Built-in Types
+
+Antelope supports several C++ data types for developing smart contracts. Developers can use these types to define data structures and write functions that interact with the EOS blockchain and smart contract system. Here are some of the most commonly used types:
+
+- `uint64_t`: This is an unsigned 64-bit integer type used for storing numeric values. It's commonly used for representing amounts of currency or other assets.
+- `name`: This is a type that represents an account name on the EOSIO network. Account names are 12-character strings that uniquely identify user accounts.
+- `asset`: This is a type that represents a quantity of a particular asset, such as EOS or another token. It includes both the amount and the symbol of the asset.
+- `string`: This is a type that represents a sequence of characters, such as a message or a username.
+- `time_point_sec`: This is a type that represents a point in time as the number of seconds since the Unix epoch (January 1, 1970).
+- `bool`: This is a type that represents a Boolean value, which can be either true or false.
+- `vector`: This is a type that represents a dynamic array of elements. It can be used to store lists of items such as account names or asset quantities.
+
+You can find a full list of all defined built-in types in the [abi_serializer.cpp file](https://github.com/AntelopeIO/leap/blob/a3e0756474a0899f94161b031a30bce4c496b292/libraries/chain/abi_serializer.cpp#L90-L128)
```

Comment:
I'd rather see a full list here instead of linking to an external code file. 

Make a nice little table with that stuff in it, and group the overlapping ones to keep it short and sweet

uintN_t -- 8/16/32/64/128
intN_t -- 8/16/32/64/128
checksum160/256/512
asset/extended

Reply:
I do not see this useful at all, especially overlapping them... take a closer look at that full list in the link provided. there's not much of overlap for the underlying types. they are all pretty much different types.

the guide already lists the most common ones, if the user really wants to see all of the and the underlying c++ type then he can consult that list. the c++ beginner prolly will not pursue the path and the c++ more advance person will not lose the path back by reading that list of eosio and c++ types.

Reply:
We shouldn't be linking to code in tutorials. Take a look at that file and try to see it from the point of view of someone who might never have touched c++ before.

They will be overwhelmed with the unnecessary code surrounding what they are actually looking for (a clear list of the types available to them) and aren't getting enough information about types they could be using.

Reply:
I added the full list